sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -94,8 +95,12 @@ sealed trait Partitioning {
    */
   def compatibleWith(other: Partitioning): Boolean
 
-  /** Returns the expressions that are used to key the partitioning. */
-  def keyExpressions: Seq[Expression]
```

Comment:
Seems `keyExpressions` is not used at all and I do not remember when we added it. So, I am removing it.